packages/react-native-windows-init/src/Cli.ts

```diff
@@ -499,9 +535,18 @@ function isProjectUsingYarn(cwd: string): boolean {
       nuGetTestVersion: argv.nuGetTestVersion,
       nuGetTestFeed: argv.nuGetTestFeed,
     });
+    return setExit(ExitCode.SUCCESS);
   } catch (error) {
-    console.error(chalk.red(error.message));
-    console.error(error);
-    process.exit(EXITCODE_UNKNOWN_ERROR);
+    const exitCode = (error instanceof UserError) ? (error as UserError).exitCode : ExitCode.UNKNOWN_ERROR;
+    if (exitCode != ExitCode.SUCCESS) 
+    {
+      console.error(chalk.red(error.message));
+      console.error(error);
+    }
+    setExit(exitCode, error.message);
+  } finally {
+    telClient.flush();
```

Comment:
You could also attach an exit handler to the process, which I think would catch the process.exit case. https://stackoverflow.com/questions/14031763/doing-a-cleanup-action-just-before-node-js-exits

Reply:
`flush` is sadly asynchronous, the problem is the telemetry client expects to continue to run all the way to the end.